### PR TITLE
XWIKI-13912: Impossible to control contrast of the panel headers on Flamingo

### PR DESCRIPTION
**Pull request by Sereza7**
## Jira
https://jira.xwiki.org/browse/XWIKI-13912

## PR Changes
* Added the panel header color to the color theme UI
* Updated variable name to fit in their new context

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm [hunk 1]
```diff
@@ -98,8 +98,8 @@
 
   @panel-bg:                            $theme.panelBackgroundColor;
   @panel-default-text:                  $theme.panelTextColor;
-  @xwiki-panel-header-bg:               $theme.panelHeaderBackgroundColor;
-  @xwiki-panel-header-text:             $theme.panelHeaderTextColor;
+  @panel-header-bg:                     $theme.panelHeaderBackgroundColor;
+  @panel-header-text:                   $theme.panelHeaderTextColor;
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm:102]: Wouldn't it be better to (additionally) keep the old variable names for backwards-compatibility?

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm [hunk 2]
```diff
@@ -225,8 +226,8 @@
   ## currently, it comes from the old color theme object (see colorThemeInit.vm)
   // Default
   @xwiki-border-color:                  $theme.borderColor;
-  @xwiki-panel-header-bg:               $theme.panelHeaderBackgroundColor;
-  @xwiki-panel-header-text:             $theme.panelHeaderTextColor;
+  @panel-header-bg:                     $theme.panelHeaderBackgroundColor;
+  @panel-header-text:                   $theme.panelHeaderTextColor;
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm:230]: Same here, wouldn't it be better to (additionally) keep the old variable names for backwards-compatibility?

Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm:233]: Addressed in https://github.com/xwiki/xwiki-platform/pull/2497/commits/558749259766abd5e3cd5e8bfbf16fceed5bca47 :+1:

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm (hunk 1 above)
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm:102]: Addressed in https://github.com/xwiki/xwiki-platform/pull/2497/commits/558749259766abd5e3cd5e8bfbf16fceed5bca47 :+1:

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm [hunk 3]
```diff
@@ -98,8 +98,11 @@
 
   @panel-bg:                            $theme.panelBackgroundColor;
   @panel-default-text:                  $theme.panelTextColor;
-  @xwiki-panel-header-bg:               $theme.panelHeaderBackgroundColor;
-  @xwiki-panel-header-text:             $theme.panelHeaderTextColor;
+  @panel-header-bg:                     $theme.panelHeaderBackgroundColor;
+  @panel-header-text:                   $theme.panelHeaderTextColor;
+  ## Kept for backward compatibility
+  @panel-header-bg:                     $theme.panelHeaderBackgroundColor;
+  @panel-header-text:                   $theme.panelHeaderTextColor;
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm:105]: Wrong less variable name, this should be `@xwiki-panel-header-bg`/`-text`.

**Comment by michitux** [comment]
@Sereza7 Could you please resolve the conflicts due to the merge of #2485?

**Comment by Sereza7** [comment]
Yes, sorry, it was a bit lengthy ^^'

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm (hunk 3 above)
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/variablesInit.vm]: Sorry, I did not check my changes properly before commiting. Fixed in https://github.com/xwiki/xwiki-platform/pull/2497/commits/dc4c9d593fead0265fddad2e214ce066f44c344b

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-theme/xwiki-platform-flamingo-theme-bootswatch/src/main/resources/FlamingoThemes/Cyborg.xml [hunk 4]
```diff
@@ -2050,6 +2063,9 @@ a.list-group-item {
     <property>
       <panel-bg>#060606</panel-bg>
     </property>
+    <property>
+      <panel-default-text>#ffffff</panel-default-text>
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-theme/xwiki-platform-flamingo-theme-bootswatch/src/main/resources/FlamingoThemes/Cyborg.xml:2067]: This looks like a bad merge.

**Review by michitux — APPROVED** [review]
Looks good overall, but there is one theme file that looks wrong, see my comment.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-theme/xwiki-platform-flamingo-theme-bootswatch/src/main/resources/FlamingoThemes/Cyborg.xml (hunk 4 above)
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-theme/xwiki-platform-flamingo-theme-bootswatch/src/main/resources/FlamingoThemes/Cyborg.xml]: Addressed in https://github.com/xwiki/xwiki-platform/pull/2497/commits/9c640c2eaae7adb96e7a584c41293f63e6e580b0 :+1:

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-theme/xwiki-platform-flamingo-theme-bootswatch/src/main/resources/FlamingoThemes/Superhero.xml [hunk 5]
```diff
@@ -2125,7 +2138,7 @@ div.main {
   box-shadow: none;
 }
 
-@xwiki-panel-header-text: @navbar-default-color;
+@panel-header-text: @navbar-default-color;
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-theme/xwiki-platform-flamingo-theme-bootswatch/src/main/resources/FlamingoThemes/Superhero.xml:2141]: Should this either a) set the corresponding theme variable or b) also set the fallback value?

Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-theme/xwiki-platform-flamingo-theme-bootswatch/src/main/resources/FlamingoThemes/Superhero.xml:2142]: For better backwards compatibility, and consistency with the choice we made in `variablesInit.vm`, I added back the former value as a fallback in https://github.com/xwiki/xwiki-platform/pull/2497/commits/afc21a6f696fc2ef84bb681067afe1ad954535b2 :+1:

**Pull request merged**